nbconverted/3_park_analysis.py

```diff
@@ -0,0 +1,377 @@
+
+
+# park geneset info
+park_loss_data = cfg.data_dir / 'park_loss_df.tsv'
+park_gain_data = cfg.data_dir / 'park_gain_df.tsv'
+
+# park gene/cancer type predictions
+park_preds_dir = cfg.data_dir / 'park_genes_preds'
+
+# mutation and copy number data
+pancancer_pickle = Path('/home/jake/research/mpmp/data/pancancer_data.pkl')
+
+
+# ### Load mutation info
+# 
+# For now, just use binary mutation status from the pancancer repo. In the future we could pull more granular info from MC3, but it would take some engineering of `1_get_mutation_counts` to do this for lots of genes.
+
+# In[3]:
+
+
+park_loss_df = pd.read_csv(park_loss_data, sep='\t', index_col=0)
+park_loss_df.head()
+
+
+# In[4]:
+
+
+park_gain_df = pd.read_csv(park_gain_data, sep='\t', index_col=0)
+park_gain_df.head()
+
+
+# In[5]:
+
+
+with open(pancancer_pickle, 'rb') as f:
+    pancancer_data = pkl.load(f)
+
+
+# In[6]:
+
+
+mutation_df = pancancer_data[1]
+print(mutation_df.shape)
+mutation_df.iloc[:5, :5]
+
+
+# ### Load copy number info
+# 
+# Get copy loss/gain info directly from GISTIC "thresholded" output. This should be the same as (or very similar to) what the Park et al. study uses.
+
+# In[7]:
+
+
+# we use the data source and preprocessing code from the pancancer repo, here:
+# https://github.com/greenelab/pancancer/blob/d1b3de7fa387d0a44d0a4468b0ac30918ed66886/scripts/initialize/process_copynumber.py#L21
+
+copy_thresh_df = (
+    pd.read_csv(Path('data', 'pancan_GISTIC_threshold.tsv'),
+                sep='\t', index_col=0)
+      .drop(columns=['Locus ID', 'Cytoband'])
+)
+copy_thresh_df.columns = copy_thresh_df.columns.str[0:15]
+
+print(copy_thresh_df.shape)
+copy_thresh_df.iloc[:5, :5]
+
+
+# In[8]:
+
+
+sample_freeze_df = pancancer_data[0]
+copy_samples = list(
+    set(sample_freeze_df.SAMPLE_BARCODE)
+    .intersection(set(copy_thresh_df.columns))
+)
+print(len(copy_samples))
+
+
+# In[9]:
+
+
+# make sure we're not losing too many samples, a few is fine
+print(set(sample_freeze_df.SAMPLE_BARCODE) - set(copy_thresh_df.columns))
+
+
+# In[10]:
+
+
+copy_thresh_df = (copy_thresh_df
+    .T
+    .loc[sorted(copy_samples)]
+    .fillna(0)
+    .astype(int)
+)
+
+print(copy_thresh_df.shape)
+copy_thresh_df.iloc[:5, :5]
+
+
+# In[11]:
+
+
+# thresholded copy number includes 5 values [-2, -1, 0, 1, 2], which
+# correspond to "deep loss", "moderate loss", "no change",
+# "moderate gain", and "deep gain", respectively.
+#
+# here, we want to use "moderate" and "deep" loss/gain to define CNV
+# loss/gain, as opposed to the more conservative approach of using
+# "deep loss/gain" as in our classifiers
+
+copy_loss_df = (copy_thresh_df
+    .replace(to_replace=[1, 2], value=0)
+    .replace(to_replace=[-1, -2], value=1)
+)
+print(copy_loss_df.shape)
+copy_loss_df.iloc[:5, :5]
+
+
+# In[12]:
+
+
+copy_gain_df = (copy_thresh_df
+    .replace(to_replace=[-1, -2], value=0)
+    .replace(to_replace=[1, 2], value=1)
+)
+print(copy_gain_df.shape)
+copy_gain_df.iloc[:5, :5]
```

Comment:
Okay, as the tables go on I'm getting more confused, not less. I think a quick description of what you expect at each step could go a long way. Currently I just look at the table and see some rows with all 0s, some with all 1s, and my mind goes "loading/parsing error"